src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) ForgeFlow, S.L. 2020
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+from .. import github
+from ..config import switchable
+from ..manifest import _user_can_push
+from ..process import CalledProcessError, check_call
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            with github.temporary_clone(org, repo, target_branch) as clone_dir:
+                if not _user_can_push(
+                    gh, org, repo, username, clone_dir, target_branch
+                ):
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username} you are not allowed to rebase.\n\n"
+                        f"To do so you must either have push permissions on "
+                        f"the repository, or be a declared maintainer of all "
+                        f"modified addons.\n\n"
+                        f"If you wish to adopt an addon and become it's "
+                        f"[maintainer]"
+                        f"(https://odoo-community.org/page/maintainer-role), "
+                        f"open a pull request to add "
+                        f"your GitHub login to the `maintainers` key of its "
+                        f"manifest.",
+                    )
+                    return
+                # get remote name
+                remote = gh_pr.head._repo_owner
+                if not remote:
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username}, impossible to rebase "
+                        f"because unknown remote.",
+                    )
+                    return
+                # fetch pr and create local branch
+                current_branch = github.git_get_current_branch(cwd=clone_dir)
```

Comment:
I'm not sure this is correct. If I read correctly, current_branch is going to be the same as target_branch, because that is what has been checked out in temporary_clone above, it has not changed since.